src/utils.js

```diff
@@ -204,9 +204,13 @@ exports.withTimeout = (asyncFn, time) => {
     return Promise.race([
       asyncFn(...args),
       new Promise((resolve, reject) => {
-        setTimeout(() => {
+        const timeout = setTimeout(() => {
           reject(errcode(new Error('Async function did not complete before timeout'), 'ETIMEDOUT'))
         }, time)
+
+        if (timeout && timeout.unref) {
```

Comment:
`setTimeout` return a number, `timeoutId`. Type checker is failing here as a consequence

Reply:
In node it returns a [Timeout](https://nodejs.org/api/timers.html#timers_class_timeout) which has an `unref` method property.  Weird, maybe we need to tell ts that it's used under node.

Reply:
I'm managed to refactor this away after removing the unnecessary async and using the `p-timeout` module instead of `Promise.race`.